### PR TITLE
Added cleaning resources

### PR DESCRIPTION
**Pull request by segabriel**

### CODE REVIEW

**Review by artem-v — COMMENTED** [review]
Plz add unit tests. It looks like that aint' hard to do because you depend only on intterfaces

File: rsocket-transport/src/main/java/io/scalecube/services/transport/rsocket/RSocketServiceTransport.java [hunk 1]
```diff
@@ -67,8 +67,12 @@ public EventLoopGroup getExecutorService() {
   }
 
   @Override
-  public Mono<Void> shutdown(ExecutorService executorService) {
-    return Mono.defer(
-        () -> FutureMono.from((Future) ((EventLoopGroup) executorService).shutdownGracefully()));
+  public Mono shutdown(ExecutorService executorService) {
+    if (executorService == null) {
```
Comment (dmytro-lazebnyi) [inline rsocket-transport/src/main/java/io/scalecube/services/transport/rsocket/RSocketServiceTransport.java]: Wrap whole method to Mono.defer?

File: services-api/src/main/java/io/scalecube/services/transport/api/ServiceTransport.java [hunk 2]
```diff
@@ -46,5 +46,5 @@ static ServiceTransport getTransport() {
    * @param executorService transport executor service
    * @return shutdown signal
    */
-  Mono<Void> shutdown(ExecutorService executorService);
+  Mono shutdown(ExecutorService executorService);
```
Comment (dmytro-lazebnyi) [inline services-api/src/main/java/io/scalecube/services/transport/api/ServiceTransport.java]: Why `Void` is removed? What are we expecting from return type `Mono`?

File: services-discovery/src/main/java/io/scalecube/services/discovery/ScalecubeServiceDiscovery.java [hunk 3]
```diff
@@ -86,8 +86,10 @@ public ServiceEndpoint endpoint() {
 
   @Override
   public Mono<Void> shutdown() {
-    sink.complete();
-    return Mono.fromFuture(cluster.shutdown());
+    return Mono.defer(() -> {
+      sink.complete();
+      return cluster != null ? Mono.fromFuture(cluster.shutdown()) : Mono.empty();
```
Comment (dmytro-lazebnyi) [inline services-discovery/src/main/java/io/scalecube/services/discovery/ScalecubeServiceDiscovery.java]: In accordance with similar changes maybe better to use `Optional`?

File: services/src/main/java/io/scalecube/services/Microservices.java [hunk 4]
```diff
@@ -322,7 +326,12 @@ private GatewayBootstrap addConfig(GatewayConfig config) {
     }
 
     private Mono<Void> shutdown() {
-      return Flux.fromIterable(gatewayInstances.values()).flatMap(Gateway::stop).then();
+      return Mono.defer(
+          () ->
+              gatewayInstances != null && !gatewayInstances.isEmpty()
```
Comment (dmytro-lazebnyi) [inline services/src/main/java/io/scalecube/services/Microservices.java:331]: In accordance with similar changes maybe better to use `Optional`?

File: services-api/src/main/java/io/scalecube/services/transport/api/ServiceTransport.java (hunk 2 above)
Comment (artem-v) [reply services-api/src/main/java/io/scalecube/services/transport/api/ServiceTransport.java]: @dmytro-lazebnyi good point, I will set it back

File: services-discovery/src/main/java/io/scalecube/services/discovery/ScalecubeServiceDiscovery.java (hunk 3 above)
Comment (artem-v) [reply services-discovery/src/main/java/io/scalecube/services/discovery/ScalecubeServiceDiscovery.java]: @dmytro-lazebnyi yep

File: services/src/main/java/io/scalecube/services/Microservices.java (hunk 4 above)
Comment (artem-v) [reply services/src/main/java/io/scalecube/services/Microservices.java:331]: @dmytro-lazebnyi nope

**Pull request merged**